lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -365,14 +380,14 @@ def main():
                 vpc_security_group_ids_changed = sorted(vpc_security_group_ids) != sorted(current_vpc_security_group_ids)
 
             if 'VpcConfig' not in current_config or subnet_net_id_changed or vpc_security_group_ids_changed:
-                func_kwargs.update({'VpcConfig':
-                                    {'SubnetIds': vpc_subnet_ids,'SecurityGroupIds': vpc_security_group_ids}})
+                new_vpc_config = {'SubnetIds': vpc_subnet_ids,
+                                  'SecurityGroupIds': vpc_security_group_ids}
+                func_kwargs.update({'VpcConfig': new_vpc_config})
         else:
-            # No VPC configuration is desired, assure VPC config is empty when present in current config
-            if ('VpcConfig' in current_config and
-                    'VpcId' in current_config['VpcConfig'] and
-                    current_config['VpcConfig']['VpcId'] != ''):
-                func_kwargs.update({'VpcConfig':{'SubnetIds': [], 'SecurityGroupIds': []}})
+            # No VPC configuration is desired, assure VPC config is empty when
+            # present in current config
+            if 'VpcConfig' in current_config and 'VpcId' in current_config['VpcConfig'] and current_config['VpcConfig']['VpcId'] != '':
```

Comment:
Can be simplified to `if 'VpcConfig' in current_config and current_config['VpcConfig'].get('VpcId'):` (since an empty string evaluates to false)